test/select-test.js

```diff
@@ -1,52 +1,45 @@
-var tape = require("tape"),
-    jsdom = require("./jsdom"),
-    d3 = require("../");
+import assert from "assert";
+import * as d3 from "../src/index.js";
+import jsdom from "./jsdom.js";
 
-tape("d3.select(…) returns an instanceof d3.selection", function(test) {
-  var document = jsdom("<h1>hello</h1>");
-  test.ok(d3.select(document) instanceof d3.selection);
-  test.end();
+it("d3.select(…) returns an instanceof d3.selection", () => {
+  const document = jsdom("<h1>hello</h1>");
+  assert(d3.select(document) instanceof d3.selection);
 });
 
-tape("d3.select(string) selects the first element that matches the selector string", function(test) {
-  var document = global.document = jsdom("<h1 id='one'>foo</h1><h1 id='two'>bar</h1>");
+it("d3.select(string) selects the first element that matches the selector string", () => {
+  const document = jsdom("<h1 id='one'>foo</h1><h1 id='two'>bar</h1>");
   try {
-    test.deepEqual(d3.select("h1"), {_groups: [[document.querySelector("h1")]], _parents: [document.documentElement]});
-    test.end();
+    assert.deepStrictEqual(d3.select("h1"), {_groups: [[document.querySelector("h1")]], _parents: [document.documentElement]});
```

Comment:
```suggestion
    assert.deepStrictEqual(d3.select("h1"), new Selection([[document.querySelector("h1")]], [document.documentElement]));
```

The reason why this does not work is because we have changed the `deepEqual` to `deepStrictEqual`, then it is much more strict on how it should assert the objects. It is trying to assert a javascript function with a javascript object, and that does not work. 

I am not sure if this is the right way of doing it though, I'm just giving an example on how to fix it. 

I've imported the `Selection` function from `./src/selection/index.js`